src/apps/auth/services/auth.service.ts

```diff
@@ -63,15 +66,40 @@ export class AuthService {
     }
 
     return {
-      data: await this._genereateToken(userAuth),
+      data: await this._generateToken(userAuth),
+    };
+  }
+
+  /**
+   * Refresh token
+   *
+   * @param {RefreshInput} data
+   * @returns {Promise<TokenOutput>}
+   */
+  async refresh(data: RefreshInput): Promise<ResourceDataOutput<TokenOutput>> {
+    const verified: any = verifyJwtToken(data.refreshToken);
+    if (!verified?.userId) {
+      const { errorCode, message, httpCode } = Errors.INVALID_REFRESH_TOKEN;
+      throw new AppException(errorCode, message, httpCode);
+    }
+
+    const token = await this.tokenRepository.findOne({
+      refreshToken: data.refreshToken,
+    });
```

Comment:
If the refresh token has been removed once the user logged out, what should we do at here? :)